### ISSUE TITLE
lazy.nvim conflicts with fnl ftplugins

### ISSUE THREAD

**Issue by fdschmidt93** [rktjmp/hotpot.nvim#112]
Hi there!

Taking my first crack at fennel and lisps in general and liking hotpot very much so far -- thank you for a convenient plugin :)

In learning fennel via writing a plugin, I found myself needing an `ftplugin`, which unfortunately didn't work.

I was able to pin it down quickly to lazy with a separate clean Neovim config: In a nutshell, it appears that fennel `ftplugins` do not work (do work) when `lazy.nvim` is (is not) included as per the instructions on the README.

More specifically steps to reproduce:

1. create `~/.config/nvim_clean`
2. Create `~/.config/nvim_clean/ftplugin/lua.fnl` with content `(print "hello")`
3. Move one of the two lua snippets into `~/.config/nvim_clean/init.lua`, depending on scenario
4. Launch shell with `export NVIM_APPNAME="nvim_clean"`
5. Open lua file and observe behavior

Works
```lua
-- Bootstap hotpot into lazy plugin dir if it does not exist yet.
local hotpotpath = vim.fn.stdpath("data") .. "/lazy/hotpot.nvim"
if not vim.loop.fs_stat(hotpotpath) then
  vim.notify("Bootstrapping hotpot.nvim...", vim.log.levels.INFO)
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    -- You may with to pin a known version tag with `--branch vX.Y.Z`
    "https://github.com/rktjmp/hotpot.nvim.git",
    hotpotpath,
  })
end
-- As per lazy's install instructions, but insert hotpots path at the front
vim.opt.runtimepath:prepend({hotpotpath})
require("hotpot").setup()
```

Doesn't work
```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    "--branch=stable", -- latest stable release
    lazypath,
  })
end
-- Bootstap hotpot into lazy plugin dir if it does not exist yet.
local hotpotpath = vim.fn.stdpath("data") .. "/lazy/hotpot.nvim"
if not vim.loop.fs_stat(hotpotpath) then
  vim.notify("Bootstrapping hotpot.nvim...", vim.log.levels.INFO)
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    -- You may with to pin a known version tag with `--branch vX.Y.Z`
    "https://github.com/rktjmp/hotpot.nvim.git",
    hotpotpath,
  })
end
-- As per lazy's install instructions, but insert hotpots path at the front
vim.opt.runtimepath:prepend({hotpotpath, lazypath})
require("hotpot").setup() -- optionally you may call require("hotpot").setup(...) here
-- include hotpot as a plugin so lazy will update it
local plugins = {"rktjmp/hotpot.nvim"}
require("lazy").setup(plugins)
```

**Comment by rktjmp** [comment]
Thanks for the report. I am in the midst of rebuilding some drives after a failure so it might take me a bit to get to this.

Can you try:

- enabling vim.loader (`vim.loader.enable()` AFAICR)
- confirm the plugins do not work after restarting nvim after trying to run the ftplugin (the vim.loader caches "top mods" it has seen and requires some nudging to recheck things, lazy might need similar nudging but restarting should do in a pinch.)

**Comment by fdschmidt93** [comment]
> confirm the plugins do not work after restarting nvim after trying to run the ftplugin (the vim.loader caches "top mods" it has seen and requires some nudging to recheck things, lazy might need similar nudging but restarting should do in a pinch.)

Not 100% sure I followed this correctly, but essentially repeatedly tried placing `vim.loader.enable()` at top / bottom of "doesn't work" config and saw no change. Let me know in case I did not properly tested what you had in mind. Didn't seem to make a difference.

I found for more evidence by inspecting the loader via

`:lua vim.print(vim.loader.find("*", { all = true }))`

"works": includes the 

```lua
{                                                                                                                                                                                 
    modname = "hotpot-ftplugin",
    modpath = "/home/fdschmidt/.cache/nvim_clean/hotpot/compiled/ftplugin-nvim_clean/lua/hotpot-ftplugin"
  },
```

while "doesn't work" does not.

**Comment by rktjmp** [comment]
Should be fixed in 0.9.1

**Comment by fdschmidt93** [comment]
Yes, thanks for the quick fix!
